sourmash/commands.py

```diff
@@ -730,108 +733,114 @@ def multigather(args):
-            average_abund ='{:.1f}'.format(result.average_abund)
-            name = result.match._display_name(40)
 
-            if query.minhash.track_abundance and not args.ignore_abundance:
-                print_results('{:9}   {:>7} {:>7} {:>9}    {}',
-                          format_bp(result.intersect_bp), pct_query, pct_genome,
-                          average_abund, name)
-            else:
-                print_results('{:9}   {:>7} {:>7}    {}',
-                          format_bp(result.intersect_bp), pct_query, pct_genome,
-                          name)
-            found.append(result)
+            # basic reporting
+            print_results('\nfound {} matches total;', len(found))
 
+            print_results('the recovered matches hit {:.1f}% of the query',
+                   (1 - weighted_missed) * 100)
+            print_results('')
 
-        # basic reporting
-        print_results('\nfound {} matches total;', len(found))
-
-        print_results('the recovered matches hit {:.1f}% of the query',
-               (1 - weighted_missed) * 100)
-        print_results('')
-
-        if not found:
-            notify('nothing found... skipping.')
-            continue
-
-        output_base = os.path.basename(queryfile)
-        output_csv = output_base + '.csv'
-
-        fieldnames = ['intersect_bp', 'f_orig_query', 'f_match',
-                      'f_unique_to_query', 'f_unique_weighted',
-                      'average_abund', 'median_abund', 'std_abund', 'name',
-                      'filename', 'md5', 'f_match_orig']
-        with open(output_csv, 'wt') as fp:
-            w = csv.DictWriter(fp, fieldnames=fieldnames)
-            w.writeheader()
-            for result in found:
-                d = dict(result._asdict())
-                del d['match']                 # actual signature not in CSV.
-                w.writerow(d)
-
-        output_matches = output_base + '.matches.sig'
-        with open(output_matches, 'wt') as fp:
-            outname = output_matches
-            notify('saving all matches to "{}"', outname)
-            sig.save_signatures([ r.match for r in found ], fp)
-
-        output_unassigned = output_base + '.unassigned.sig'
-        with open(output_unassigned, 'wt') as fp:
             if not found:
-                notify('nothing found - entire query signature unassigned.')
-            elif not len(query.minhash):
-                notify('no unassigned hashes! not saving.')
-            else:
-                notify('saving unassigned hashes to "{}"', output_unassigned)
+                notify('nothing found... skipping.')
+                continue
+
+            # need to handle this better/differently when query.filename not properly set
+            query_filename = query.filename
+            if not query_filename:
+                query_filename = queryfile
+
+            output_base = os.path.basename(query_filename)
+            output_csv = output_base + '.csv'
```

Comment:
what is the best way to set an output base when querying using an index file? Or is query.filename always set properly now?

Reply:
ugh, I don't know what to do here :). I think my original idea was that multigather would be used with a collection of files that contained individual signatures... it's probably better to use information specific to the signature tho! A couple thoughts --

* you could use the md5sum of the query.
* you could use the name or filename of the query (but query.filename is, I think, the name of the FASTA file the query was built from, so...)
* ...no other good ideas.

md5sum is probably the safest, at least to start.

ooh, alternatively, you could build a unique name off of the queryfile by using the number of the signature that you're using as query, e.g. `<queryfile>.0.csv` for the first one, `<queryfile>.1.csv`, etc.

Reply:
I think my suggestion is: get it working with md5sum, we can revisit :)